content/en/docs/components/pipelines/author-a-pipeline/components.md

```diff
@@ -0,0 +1,225 @@
++++
+title = "Components"
+description = "Author KFP components"
+weight = 4
++++
+
+
+## Summary
+A *component* is the basic unit of execution logic in KFP. A component is a named template for how to run a container using an image, a command, and arguments. Components may also have inputs and outputs, making a component a computational template, analogous to a function.
+
+Component inputs are dynamic data used in either the container commands or arguments.
+
+Component outputs may be machine learning artifacts or other JSON-serializable data.
+
+## Author a component
+
+At the lowest level of execution, all components define their execution logic via a container image, command, and arguments. An importer component is a special case and the only exception to this.
+
+The KFP SDK exposes three ways of authoring components with these three properties.
+
+### 1. Lighweight Python function-based components
+The most simple way to author a component is via a lightweight Python function-based component (also known as a lightweight component).
+
+Lightweight components provides a fully Pythonic approach to creating a component that executes a single Python function within a container at runtime.
+
+To create a lightweight component, you must:
+1. Define a standalone function.
+
+    A standalone Python function is a function that does not reference any symbols defined outside of its scope. This means the function must define all objects it uses within the scope of the function and must include all import statements within the function body.
+
+2. Include type annotations for the function parameters and return values.
+
+    Type annotations indicate what the component inputs and outputs are and tells the KFP lightweight component executor how to serialize and deserialize the data as it is passed within a pipeline. This also (optionally) allows the KFP DSL compiler to type check your pipeline.
+
+    Valid parameter annotations include `str`, `int`, `float`, `bool`, `dict`, `list`, `OutputPath`, `InputPath`, `Input[<Artifact>]`, and `Output[<Artifact>]`.
```

Comment:
We also support `InputUri` and `OutputUri`, but didn't advertise these as we prefer `Input` and `Output` which covers both path and uri.

Reply:
I know we have these placeholders, but do we have an annotation to indicate these? Or are they only accessible via the `artifact.uri` notation in container components?

Reply:
We have those available to use as placeholders in v1 component yaml: https://github.com/kubeflow/pipelines/blob/f17e9625dbe0cfad1e7a7e87fa7cccc79e08c2ac/sdk/python/kfp/v2/compiler_cli_tests/test_data/two_step_pipeline.py#L53

but not as annotations--change was prepared but discarded: https://github.com/kubeflow/pipelines/pull/5634

So I think maybe we can skip these in the doc as well, for now.